dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/importer/EventManager.java

```diff
@@ -121,9 +124,10 @@ public ImportSummaries addEvents( final List<Event> events, final WorkContext wo
         // pre-process events
         executorsByPhase.get( EventProcessorPhase.INSERT_PRE ).execute( workContext, validEvents );
```

Comment:
I am starting to wonder if the pro-processing here should be within the follwoing `if` blocks for each of the types. Not really related to the PR changes so far but something maybe worth thinking about. We might be able to avoid doing work.